### PR TITLE
Refactor: migrate v2 forms to campaigns and add upgraded forms as campaign not-default forms

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-2032]

## Description

This PR makes the migrations of Forms to Campaigns also consider V2 forms. It also fixed a problem that was preventing forms with the upgraded status from being associated with campaigns as not-default forms in certain scenarios.

## Affects

The migrations of Forms to Campaigns.

## Visuals

https://www.loom.com/share/7bf0d4727f944c0da29fd0df9219d7bc?sid=2b9f341f-c61a-4f13-b088-b8600b09afab

## Testing Instructions

1. In a fresh install create a few V2 and V3 forms;
2. Migrate some V2 forms to V3 forms;
3. Install the Give version of this branch;
4. The migration should convert V2 forms to Campaigns;
5. The migration should associate V2 upgraded forms to Campaigns as not-default forms.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-2032]: https://stellarwp.atlassian.net/browse/GIVE-2032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Campaigns/Migrations/MigrateFormsToCampaignForms.php [hunk 1]
```diff
@@ -77,17 +77,43 @@ protected function getFormData(): array
...
+         * Exclude forms with an `upgraded` status, which are WP revisions.
+         *
+         * @see https://wordpress.org/documentation/article/post-status/#auto-draft
+         */
+        $query->where('forms.post_status', 'auto-draft', '!=');
+
+        /**
+         * Excluded upgraded V2 forms as their corresponding V3 version will be used to create the campaign - later the V2 form will be added to the proper campaign as a non-default form through the addUpgradedV2FormToCampaign() method.
+         */
+        $upgradedFormsIds = $this->getUpgradedV2FormsIds();
```
Comment (kjohnson) [inline src/Campaigns/Migrations/MigrateFormsToCampaignForms.php:90]: Could this be refactored to a join of the meta with a "where not"? Or do we need a second query?

File: src/Campaigns/Migrations/MigrateFormsToCampaignForms.php [hunk 2]
```diff
@@ -77,17 +77,43 @@ protected function getFormData(): array
...
             ->whereIsNull('campaigns.id');
 
-        // Exclude forms with an `upgraded` status, which are archived.
-        $query->where('forms.post_status', 'upgraded', '!=');
+        /**
+         * Exclude forms with an `upgraded` status, which are WP revisions.
+         *
+         * @see https://wordpress.org/documentation/article/post-status/#auto-draft
+         */
+        $query->where('forms.post_status', 'auto-draft', '!=');
```
Comment (kjohnson) [inline src/Campaigns/Migrations/MigrateFormsToCampaignForms.php:85]: I think this is conflating two different things. We do need to exclude `auto-draft` as a status, but I don't think that relates in any way to a form being "upgraded" (as stated in the comment above).

File: tests/Unit/Campaigns/Migrations/MigrateFormsToCampaignFormsTest.php [hunk 3]
```diff
@@ -19,6 +20,8 @@ final class MigrateFormsToCampaignFormsTest extends TestCase
 
     /**
      * @unreleased
+     *
+     * @throws Exception
      */
     public function testCreatesParentCampaignForDonationForm()
```
Comment (kjohnson) [inline tests/Unit/Campaigns/Migrations/MigrateFormsToCampaignFormsTest.php:26]: @glaubersilva can you add a new test to cover V2 forms that are not upgraded?

File: src/Campaigns/Migrations/MigrateFormsToCampaignForms.php (hunk 1 above)
Comment (glaubersilva) [reply src/Campaigns/Migrations/MigrateFormsToCampaignForms.php]: I didn't use join but I refactored it to remove the getUpgradedV2FormsIds method and kept only one query.

Resolved in b1038647d440fcd743c6bde63a4e4a6c96995b46

File: src/Campaigns/Migrations/MigrateFormsToCampaignForms.php (hunk 2 above)
Comment (glaubersilva) [reply src/Campaigns/Migrations/MigrateFormsToCampaignForms.php:86]: Resolved in f40ab332e726bc2de3f7104e59a001a506099864

File: tests/Unit/Campaigns/Migrations/MigrateFormsToCampaignFormsTest.php (hunk 3 above)
Comment (glaubersilva) [reply tests/Unit/Campaigns/Migrations/MigrateFormsToCampaignFormsTest.php:26]: Ok, I think I better understand what you said after I wrote my last comment! 😅

Resolved in eac91275f88f3f74d95e4dc43b4bde48c1f28472

**Comment by glaubersilva** [comment]
@kjohnson This is ready for your re-review.

**Review by kjohnson — APPROVED** [review]
Thanks for considering the feedback, @glaubersilva.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**